src/client_side_reply.cc

```diff
@@ -1031,74 +942,53 @@ clientReplyContext::purgeRequest()
     /* Release both IP cache */
     ipcacheInvalidate(http->request->url.host());
 
-    if (!http->flags.purging)
-        purgeRequestFindObjectToPurge();
-    else
-        purgeDoMissPurge();
-}
-
-void
-clientReplyContext::purgeDoMissPurge()
-{
-    http->logType.update(LOG_TCP_MISS);
-    lookingforstore = 3;
-    StoreEntry::getPublicByRequestMethod(this,http->request, Http::METHOD_GET);
+    // TODO: can we use purgeAllCached() here instead?
+    purgeDoPurge();
 }
 
 void
-clientReplyContext::purgeDoPurgeGet(StoreEntry *newEntry)
-{
-    if (newEntry) {
-        /* Release the cached URI */
-        debugs(88, 4, "clientPurgeRequest: GET '" << newEntry->url() << "'" );
-#if USE_HTCP
-        neighborsHtcpClear(newEntry, http->request, HttpRequestMethod(Http::METHOD_GET), HTCP_CLR_PURGE);
-#endif
-        newEntry->release(true);
-        purgeStatus = Http::scOkay;
+clientReplyContext::purgeDoPurge()
+{
+    auto firstFound = false;
+    if (const auto entry = storeGetPublicByRequestMethod(http->request, Http::METHOD_GET)) {
+        // special entries are only METHOD_GET entries without variance
+        if (EBIT_TEST(entry->flags, ENTRY_SPECIAL)) {
+            http->logType.update(LOG_TCP_DENIED);
+            Ip::Address tmp_noaddr;
+            tmp_noaddr.setNoAddr(); // TODO: make a global const
```

Comment:
This TODO can be implemented easily:
```suggestion
            static Ip::Address tmp_noaddr = Ip::Address::v6_noaddr;
```

Reply:
The above suggestion does not look like a good implementation of this out-of-scope TODO to me, but if we can indeed use Ip::Address::v6_noaddr in this context, then I would not violently object to properly implementing this TODO, even though it will also require changing clientBuildError() profile to accept Ip::Address::v6_noaddr.

Reply:
>  I would not violently object to properly implementing this TODO

Done at 7c72c484.